android/quest/src/main/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensions.kt

```diff
@@ -166,16 +166,14 @@ fun List<ActionConfig>.handleClickEvent(
       ApplicationWorkflow.DEVICE_TO_DEVICE_SYNC -> startP2PScreen(navController.context)
       ApplicationWorkflow.LAUNCH_MAP -> {
         val mapFragmentDestination = MainNavigationScreen.GeoWidgetLauncher.route
-        val isMapFragmentExists =
-          navController.currentBackStack.value.any { it.destination.id == mapFragmentDestination }
+        val isMapFragmentExists = navController.destinationExistsInBackstack(mapFragmentDestination)
         if (isMapFragmentExists) {
           navController.popBackStack(mapFragmentDestination, false)
-        } else {
-          navController.navigate(
-            resId = mapFragmentDestination,
-            args = bundleOf(NavigationArg.GEO_WIDGET_ID to actionConfig.id),
-          )
         }
+        navController.navigate(
+          resId = mapFragmentDestination,
+          args = bundleOf(NavigationArg.GEO_WIDGET_ID to actionConfig.id),
+        )
```

Comment:
**Note:** I've refactored the if else condition here such that now if the map fragment exists it is popped from the backstack and the controller navigates to the same(new) fragment. 

If this was not the intention then we can revert this part, let me know cc @dubdabasoduba @aurangzaibumer

Reply:
See my commit here https://github.com/opensrp/fhircore/pull/3283/commits/d703252255d0f27d29a7a8ce23b5cf5e16e93340/ 

This doesn't resolve the solution entirely though. It seems that it needs further investigation. Navigating from the profile back to the map exits the app instead of returning to the map's fragment.